sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/OrderbyDocumentQueryTest.java

```diff
@@ -159,6 +163,22 @@ public void queryOrderBy(String sortOrder) throws Exception {
         validateQuerySuccess(queryObservable.byPage(pageSize), validator);
     }
 
+    @Test(groups = { "query" }, timeOut = TIMEOUT)
+    public void queryOrderByRoundTrips() {
+        String query = "SELECT c.v FROM c where c.type='testing' order by c.v asc";
+        CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
+
+        int pageSize = 5;
+        CosmosPagedFlux<RoundTripDocument> queryObservable = roundTripsContainer.queryItems(query, options, RoundTripDocument.class);
+        FeedResponse<RoundTripDocument> feedResponse = queryObservable.byPage(pageSize).take(1).blockFirst();
```

Comment:
how do we verify there is only 1 query request being sent to the backend?

Reply:
True it is possible without the change that the test might pass. It should be max 4 requests for this test. One for each partition and one for each prefetch per partition. To make it more robust, I do it 100 times and I'm checking the outputDocuments to be max 10 (2 pages) and the requests captured in the diagnostics to be max 4. With these changes, the test will be unlikely to pass without the change.